girder/api/v1/user.py

```diff
@@ -312,28 +319,24 @@ def generateTemporaryPassword(self, params):
         self.requireParams('email', params)
         email = params['email'].lower().strip()
 
-        users = self.model('user').find({'email': email})
+        user = self.model('user').findOne({'email': email})
 
-        if not users.count():
+        if not user:
             raise RestException('That email is not registered.')
-        for user in users:
-            token = self.model('token').createToken(None, days=1, scope=(
-                TokenScope.USER_AUTH, TokenScope.TEMPORARY_USER_AUTH))
-            token['userId'] = user['_id']
-            self.model('token').save(token)
-            base = cherrypy.request.base.rstrip('/')
-            altbase = cherrypy.request.headers.get('X-Forwarded-Host', '')
-            if altbase:
-                base = '%s://%s' % (cherrypy.request.scheme, altbase)
-            url = '%s/#useraccount/%s/token/%s' % (
-                base, str(user['_id']), str(token['_id']))
-
-            html = mail_utils.renderTemplate('temporaryAccess.mako', {
-                'url': url,
-                'token': str(token['_id'])
-            })
-            mail_utils.sendEmail(to=email, subject='Girder: Temporary Access',
-                                 text=html)
+
+        token = self.model('token').createToken(
+            user, days=1, scope=TokenScope.TEMPORARY_USER_AUTH)
+
+        emailHost = self.model('setting').get(SettingKey.EMAIL_HOST)
```

Comment:
Shouldn't you use [``getEmailUrlPrefix``](https://github.com/girder/girder/blob/master/girder/utility/mail_utils.py#L33) here?